File: rest_client/SwitchRequest.cpp

#include "SwitchRequest.h"

SwitchRequest::SwitchRequest(String method, SwitchType type, bool isTurningOn) :
    type(type),
    isTurningOn(isTurningOn),
    Request(method, "/api/plants/logs/switch")
{}


void SwitchRequest::setBody() {
    String type = this->type == SwitchType::WaterPump ? "water_pump" : "led";

    this->body = "{\"log_type\": " + type + ", \"turning_on\": " + isTurningOn + ", \"turning_off: \"" + !isTurningOn + "}";
}

File: system_control/switch_request.py
import request

class SwitchRequest(request.Request):

    waterPump = "water_pump"
    lights = "lights"

    def __init__(self, method, type, isTurningOn):
        self.type = type
        self.isTurningOn = isTurningOn
        super().__init__(method, "/plants/switches")

    def setBody(self):
        self.body = {"log_type": self.type, "turning_on": self.isTurningOn, "turning_off": not self.isTurningOn}